matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/SessionListeners.kt

```diff
@@ -42,18 +38,19 @@ internal class SessionListeners @Inject constructor(
         }
     }
 
-    fun dispatch(block: (Session, Session.Listener) -> Unit) {
+    fun dispatch(session: Session, block: (Session, Session.Listener) -> Unit) {
```

Comment:
this is the fix, instead of relying on `sessionManager.getSessionComponent(sessionId)?.session()` which may create a new session if one doesn't exist, the callers provide a known session instance